draft-wood-tls-external-psk-importer.md

```diff
@@ -97,20 +100,37 @@ supported HashAlgorithm 'hash', the importer constructs an ImportedIdentity stru
    } ImportedIdentity;
 ~~~
 
+[[TODO: An alternative deisgn might combine label and hash into the same field so that future
+protocols which don't have a notion of HashAlgorithm don't need this field.]]
+
+ImportedIdentity.label MUST be bound to the protocol for which the key is imported. Thus,
+TLS 1.3 and QUICv1 {{!I-D.ietf-quic-transport}} MUST use "tls13" as the label. Similarly, TLS 1.2 and 
+all prior TLS versions should use "tls12" as ImportedIdentity.label, as well as SHA256 as ImportedIdentity.hash.
+Note that this means future versions of TLS will increase the number of PSKs derived from an external PSK.
+
 A unique and imported PSK (IPSK) with base key 'ipskx' bound to this identity is then computed as follows:
 
 ~~~
    epskx = HKDF-Extract(0, epsk)
-   ipskx = HKDF-Expand-Label(epskx, "derived psk", Hash(ImportedIdentity), Hash.length)
+   ipskx = HKDF-Expand-Label(epskx, "derived psk",
+                             Hash(ImportedIdentity), CipherSuite.length)
 ~~~
 
+CipherSuite.length is the length of the key necessary for the corresponding ciphersuite. For TLS 1.3,
+CipherSuite.length = Hash.length.
```

Comment:
Hrm. This reads to me a little funny since it's no longer immediately obvious that all cipher suites that share a hash also share a length. I think my comment might have made more sense in the combined label alternate design above. In that interpretation, the importer specifies label + desired size.

Maybe keep it as Hash.length right now and leave this with the other TODO above? (I think I actually used ImportedIdentity.hash.length in the universal PSKs draft, but I don't think it actually matters one way or another.)

Reply:
That works for me, too!